avm/res/insights/private-link-scope/main.bicep

```diff
@@ -0,0 +1,292 @@
+metadata name = 'Azure Monitor Private Link Scopes'
+metadata description = 'This module deploys an Azure Monitor Private Link Scope.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the private link scope.')
+@minLength(1)
+param name string
+
+@description('''Optional. Specifies the access mode of ingestion or queries through associated private endpoints in scope. For security reasons, it is recommended to use PrivateOnly whenever possible to avoid data exfiltration.
+
+  * Private Only - This mode allows the connected virtual network to reach only Private Link resources. It is the most secure mode and is set as the default when the `privateEndpoints` parameter is configured.
+  * Open - Allows the connected virtual network to reach both Private Link resources and the resources not in the AMPLS resource. Data exfiltration cannot be prevented in this mode.''')
+param accessModeSettings accessModeType
+
+@description('Optional. The location of the private link scope. Should be global.')
+param location string = 'global'
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. Configuration details for Azure Monitor Resources.')
+param scopedResources scopedResourceType
+
+@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+param privateEndpoints privateEndpointType
+
+@description('Optional. Resource tags.')
+param tags object?
+
+@sys.description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+var builtInRoleNames = {
+  'Application Group Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'ca6382a4-1721-4bcf-a114-ff0c70227b6b')
```

Comment:
Ideally, as per https://azure.github.io/Azure-Verified-Modules/specs/bicep/#id-bcpnfr5---category-composition---role-assignments-role-definition-mapping-limits we should limit the number of roles in this list to the compulsiry ones and the ones applying the most or strictly related to the primary resource type

Reply:
I have trimmed down the RBAC roles as per guidance as it was above 15. I also used our CARML Role assignment script to generate built-in roles. However seems that something has changed from the main API side where we are getting a lot more noise and unrelated roles. I had a chat to @alex about this too. I've trimmed it down to `best judgement`.